ThreeBotPackages/threefold/tfgrid_workloads/actors/workload_manager.py

```diff
@@ -201,29 +209,41 @@ def reservations_list(self, node_id, state, epoch, schema_out):
         ```
 
         ```out
-        reservations = (LO) !tfgrid.reservation.1
+        workloads = (LO) !tfgrid.reservation.workload.1
         ```
         """
         query = None
-        if node_id:
+        if node_id != INT_NULL_VALUE:
             query = self.reservation_model.IndexTable.node_id == node_id
 
         result = self.reservation_model.IndexTable.select().where(query).execute()
         reservations_ids = set([item.reservation_id for item in result])
 
-        reservations = []
+        workloads = []
         for reservation_id in reservations_ids:
             reservation = self._reservation_get(reservation_id)
-            if state and state.upper() != reservation.next_action:
+            if state and state.upper() != str(reservation.next_action):
                 continue
 
-            if epoch and epoch < reservation.epoch:
+            if epoch != INT_NULL_VALUE and epoch >= reservation.epoch:
                 continue
 
-            reservations.append(reservation)
-        return reservations
+            workload_id = 1
```

Comment:
here again, we should not deal with workload_id our self. but just copy what is in the reservation object.